jerry-debugger/jerry-client-ws.html

```diff
@@ -0,0 +1,949 @@
+  }
+
+  this.setBreakpoint = function(str)
+  {
+    line = /^(.+):([1-9][0-9]*)$/.exec(str);
+
+    if (line)
+    {
+      var functionList = lineList.get(line[2]);
+
+      for (var i = 0; i < functionList.length; ++i)
+      {
+        var func = functionList[i];
+        var resource = func.resource;
+
+        if (resource == line[1]
+            || resource.endsWith("/" + line[1])
+            || resource.endsWith("\\" + line[1]))
+        {
+          insertBreakpoint(func.lines[line[2]]);
+        }
+      }
+    }
+    else
+    {
+      for (var i in functions)
+      {
+        var func = functions[i];
+
+        if (func.name == str && func.firstLine >= 0)
+        {
+          insertBreakpoint(func.lines[func.firstLine]);
+        }
+      }
+    }
+  }
+
+  this.deleteBreakpoint = function(index)
+  {
+    breakpoint = activeBreakpoints[index];
+
+    if (!breakpoint)
+    {
+      appendLog("No breakpoint found with index " + index);
+      return;
+    }
+
+    assert(breakpoint.activeIndex == index);
+
+    delete activeBreakpoints[index];
+    breakpoint.activeIndex = -1;
+
+    var values = [ JERRY_DEBUGGER_UPDATE_BREAKPOINT,
+                   0,
+                   breakpoint.func.byte_code_cp,
+                   breakpoint.offset ];
+
+    encodeMessage("BBCI", values);
+
+    appendLog("Breakpoint " + index + " is deleted.");
+  }
+
+  this.listBreakpoints = function()
+  {
+    appendLog("List of active breakpoints:");
+    var found = false;
+
+    for (var i in activeBreakpoints)
+    {
+      appendLog("  breakpoint " + i + " at " + breakpointToString(activeBreakpoints[i]));
+      found = true;
+    }
+
+    if (!found)
+    {
+      appendLog("  no active breakpoints");
+    }
+  }
+
+  this.dump = function()
+  {
+    for (var i in functions)
+    {
+      var func = functions[i];
+      var resource = func.resource;
+
+      if (resource == '')
+      {
+        resource = "<unknown>";
+      }
+
+      appendLog("Function 0x" + Number(i).toString(16) + " '" + func.name + "' at " + resource + ":" + func.firstLine);
+
+      for (var j in func.lines)
+      {
+        var active = "";
+
+        if (func.lines[j].active >= 0)
+        {
+          active = " (active: " + func.lines[j].active + ")";
+        }
+
+        appendLog("  Breatpoint line: " + j + " at memory offset: " + func.lines[j].offset + active);
+      }
+    }
+  }
+}
+
+function debuggerCommand(event)
+{
+  if (event.keyCode != 13)
+  {
+    return true;
+  }
+
+  var command = commandBox.value.trim();
+
+  args = /^([a-zA-Z]+)(?:\s+([^\s].*)|)$/.exec(command);
+
+  if (!args)
+  {
+    appendLog("Invalid command");
+    document.getElementById("command").value = "";
+    return true;
+  }
+
+  if (!args[2])
+  {
+    args[2] = "";
+  }
+
+  if (args[1] == "help")
+  {
+    appendLog("Debugger commands:\n" +
+              "  connect IP - connect to server\n" +
```

Comment:
I think it would be great to mark the parameters to the commands explicitly, e.g. `connect <IP address>`, `break <function name>|<file name>:<line>`, etc.